zebra-chain/src/sapling/keys.rs

```diff
@@ -472,9 +472,30 @@ pub struct AuthorizingKey(pub(crate) redjubjub::VerificationKey<SpendAuth>);
 
 impl Eq for AuthorizingKey {}
 
-impl From<[u8; 32]> for AuthorizingKey {
-    fn from(bytes: [u8; 32]) -> Self {
-        Self(redjubjub::VerificationKey::try_from(bytes).unwrap())
+impl TryFrom<[u8; 32]> for AuthorizingKey {
+    type Error = &'static str;
+
+    /// Convert an array into an AuthorizingKey.
+    ///
+    /// Returns an error if the encoding is malformed or if [it does not encode
+    /// a prime-order point][1]:
+    ///
+    /// > When decoding this representation, the key MUST be considered invalid
+    /// > if abst_J returns ⊥ for either ak or nk, or if ak not in J^{(r)*}
```

Comment:
One interesting thing about the spec is that this seems to be a specification conformance rule (security rule?) but not explicitly or necessarily a consensus rule (as identified like so:

![image](https://user-images.githubusercontent.com/552961/147000049-e320f487-882f-4fcc-a1da-5d81195dfeb9.png)

)

Do we want to annotate these any more/similarly as we annotate consensus rules? What we have in these comments may be fine, just a thought